crates/cargo-wdk/src/actions/package/tests.rs

```diff
@@ -2612,7 +2807,18 @@ impl TestSetupPackageExpectations for TestPackageAction {
                 .as_ref()
                 .expect("cargo metadata must be available"),
         )
-        .expect("Wdk metadatat must be available");
+        .expect("Wdk metadata must be available");
+
+        let expected_arch = if self.target_arch.is_some() {
+            self.target_arch
+                .as_ref()
+                .expect("target arch must be available")
+                .to_string()
+        } else {
+            detect_arch_from_rustup_toolchain()
```

Comment:
we should not try to call this method in test, rather we should try to test this method itself.